modules/ROOT/pages/as2/as2-connectorMule4.adoc

```diff
@@ -0,0 +1,187 @@
+|Self config - AS2 OID| Sender partner’s OID.
+|Self config - x509 alias| Sender’s Certificate and/or private key alias in the keystore.
+|Self config - Email| Sender partner’s email.
+|Partner config - AS2 ID| Receiver partner’s OID.
+|Partner config - x509 alias| Receiver’s Certificate and/or private key alias in the keystore.
+|Partner config - Email| Receiver partner’s email.
+|HTTP Listener config| Reference to HTTP Listener global config.
+|Keystore config - Keystore Path| The path of the keystore where certificates and keys are provided.
+|Keystore config - Keystore Password| The password for the keystore.
+|Keystore config - Private key Password| If the private key requires a password.
+|===
+
+=== Configure an AS2 Listener for receiving a message
+
+. Place the connector’s `AS2 Listener` as a source in your flow as applicable for your use case and reference the global config as defined here.
++
+[%header%autowidth.spread]
+|===
+|Key| Description
+|Self config - AS2 OID| Receiver partner’s OID.
+|Self config - x509 alias| Receiver alias as defined in the keystore file.
+|Self config - Email| Receiver partner’s email.
+|Partner config - AS2 ID| Sender partner’s OID.
+|Partner config - x509 alias| Sender alias as defined in the keystore.
+|Partner config - Email| Sender partner’s email.
+|HTTP Listener config| Reference to HTTP Listener global config.
+|Keystore config - Keystore Path| The path of the keystore where certificates and keys are provided.
+|Keystore config - Keystore Password| The password for the keystore.
+|Keystore config - Private key Password| If the private key requires a password.
+|===
++
+. Configure a connection to the partner if the MDN request is to be sent asynchronously.
+. If you want to configure other connection details such as TLS, proxy, timeouts do that now.
+
+== Runtime Configuration
+
+The AS2 Connector reads data using repeatable streams. Data is read into memory until a configurable threshold is reached, at which point the data will be written to disk.
+The default threshold for writing to disk is 64KB. This can be changed through the System property `com.mulesoft.connectors.as2.CachedOutputStream.Threshold`. This property requires an Integer value:
+
+** A positive value defines the maximum amount of data to read into memory in KB.
+
+** A zero value means that all data is written directly to disk.
+
+** A value of -1 means that the data is never written to disk, and will be stored into memory.
+
+By default, the data is written to temporary files in the directory set by `java.io.tmpdir`.
+To change the directory, set the System property: `com.mulesoft.connectors.as2.CachedOutputStream.OutputDirectory`
+
+== Migrating from the AS2 connector Version 4.x
+
+The AS2 version 5 connector requires the `Async MDN` to be handled as a separate flow unlike v4.x . The retry of AS2 Message Send is not automatic. The application has to handle the retry upon failure in MDN or failure to receive MDN.
```

Comment:
```suggestion
AS2 Connector 5 requires that `Async MDN` be handled as a separate flow. The retry of the AS2 Message Send operation is no longer automatic in the flow, and the application is responsible for retrying upon failure in MDN or upon failure to receive MDN.
```
BUT...
Could this be reworded to tell the audience what IT needs to do to migrate?
Also, how can MDN (Microsoft Dev. Network) "fail"? Is it really AS2 MDN that fails?